codespell_lib/tests/test_basic.py

```diff
@@ -455,6 +455,25 @@ def test_context(tmpdir, capsys):
     assert 'ERROR' in lines[0]
 
 
+def test_ignore_regex_flag(tmpdir, capsys):
+    """Test ignore regex flag functionality."""
+    d = str(tmpdir)
+
+    # Invalid regex.
+    code, stdout, _ = cs.main('--ignore-regex=(', std=True)
+    assert code == EX_USAGE
+    assert 'usage:' in stdout
+
+    with open(op.join(d, 'flag.txt'), 'w') as f:
+        f.write('# Please see http://example.com/abandonned for info\n')
+    # Non-matching regex results in nothing being ignored.
+    assert cs.main(f.name, '--ignore-regex=^$') == 1
+    # A word can be ignored.
+    assert cs.main(f.name, '--ignore-regex=abandonned') == 0
+    # Ignoring part of the word can result in odd behavior.
+    assert cs.main(f.name, '--ignore-regex=nn') == 0
```

Comment:
Perhaps test with your default no option flag? `--ignore-regex` Or does OptParser catch that?

Reply:
It's handled by argparse. I'm a bit wary of catching SystemExit, which is what it's throwing. Per https://docs.python.org/3/library/argparse.html#exiting-methods, the right way to prevent exit from argparse is to subclass. That seems like it'd take a fair bit of wiring, though; are you comfortable relying on argparse handling rather than tests here?

Reply:
Are you sure? Can you not do a `--verbose` or `--verbose=5` type argument? If it definitely complains when you don't pass an option, I'm happy to not deal with testing argparse, although it would still be good to test with a blank one, so `--ignore-regex=""` which should map to the empty string I think, hence false.

Reply:
> Are you sure? Can you not do a `--verbose` or `--verbose=5` type argument?

Yes, and you can actually see this already with other flags like `codespell -D`.

If both with-argument and no-argument forms were supported simultaneously, `codespell --verbose foo.txt` would be an ambiguous parse: the `foo.txt` could either be an argument to `--verbose` (with-argument `--verbose=foo.txt`) or a file argument to `foo` (no-argument `--verbose`).

Behavior like `-vvv` is used to address this use-case, repeating the flag to indicate a different level (in argparse, this is `action="count"` -- https://docs.python.org/3/library/argparse.html#action).

> If it definitely complains when you don't pass an option, I'm happy to not deal with testing argparse, although it would still be good to test with a blank one, so `--ignore-regex=""` which should map to the empty string I think, hence false.

Added.